src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java

```diff
@@ -72,28 +78,36 @@ public static synchronized TranscriptParser getParser() {
    * datastore.
    */
   public void parseAndStoreTranscript(String videoId, Key lectureKey) throws IOException {
-    Document document = getTranscriptXmlAsDocument(videoId).get();
+    URL url = getTranscriptUrlForVideo(videoId);
+    Document document = fetchUrlAsXmlDocument(url);
     putTranscriptLinesInDatastore(lectureKey, document);
   }
 
+  private URL getTranscriptUrlForVideo(String videoId) throws IOException {
+    try {
+      URIBuilder urlBuilder = new URIBuilder(API_URL);
+      urlBuilder.addParameter(API_PARAM_LANG, API_LANG_ENGLISH);
+      urlBuilder.addParameter(API_PARAM_VIDEO, videoId);
+      return urlBuilder.build().toURL();
+    } catch (URISyntaxException | MalformedURLException e) {
+      throw new IOException(e.getCause());
```

Comment:
Why are you throwing a different type of exception?

Reply:
https://effectivejava.googleplex.com/third-edition#page=323